### PR TITLE
Allow to use api version with Facebook authorization

### PR DESCRIPTION
**Pull request by mikenai**
Related to issue [377](https://github.com/mobolic/facebook-sdk/issues/377). This changes allow to use scopes which specific to API version such as `business_management`(scope available only in v2.9)

### CODE REVIEW

**Comment by mikenai** [comment]
@martey I will fix all the thing acording to your feedback and then update PR.

**Comment by mikenai** [comment]
thanks for your respons and comment with details.

**Comment by mikenai** [comment]
@martey please review recent changes.

What has been done:

- `auth_url` moved to `GraphAPI.get_auth_url`
- test case "TestAuthURL" was updated to check url with default version
- I have made docs review and saw what it is already mean what user use method of `GrapAPI` entity. Because var `graph` refers to `GraphAPI` object at the begin of api doc page. So I just edit method name in doc.
```
app_id = 1231241241
canvas_url = 'https://domain.com/that-handles-auth-response/'
perms = ['manage_pages','publish_pages']
fb_login_url = graph.get_auth_url(app_id, canvas_url, perms)
print(fb_login_url)
```

File: facebook/__init__.py [hunk 1]
```diff
@@ -362,6 +364,18 @@ def debug_access_token(self, token, app_id, app_secret):
         }
         return self.request(self.version + "/" + "debug_token", args=args)
 
+    def get_auth_url(self, app_id, canvas_url, perms=None, **kwargs):
+        """
+        Build link to make login though Facebook Open Authorization (OAuth).
```
Comment (martey) [inline facebook/__init__.py]: I think this docstring is a bit confusing. Something like "Build a URL to create an OAuth dialog" or just linking to the relevant part of Facebook's documentation would be better.

File: test/test_facebook.py [hunk 2]
```diff
@@ -31,6 +31,7 @@ class FacebookTestCase(unittest.TestCase):
     empty list for test users.
 
     """
+
```
Comment (martey) [inline test/test_facebook.py]: Having an extra space between docstrings and method code isn't a convention of this project. Even if it was, stylistic changes like this should be submitted in their own pull request.

File: test/test_facebook.py [hunk 3]
```diff
@@ -151,14 +154,22 @@ def test_invalid_format(self):
 
 class TestAuthURL(FacebookTestCase):
     def test_auth_url(self):
+        default_version = facebook.VALID_API_VERSIONS[0]
+        api = facebook.GraphAPI(version=default_version)
```
Comment (martey) [inline test/test_facebook.py]: It's not necessary to pull a version from `VALID_API_VERSIONS` here - if you initialize a `GraphAPI` object without specifying a version, we already use `VALID_API_VERSIONS[0]` - see https://github.com/mobolic/facebook-sdk/blob/08bb2773cd69fe8b996d0dc10f811fbbd21ca945/facebook/__init__.py#L104.

File: facebook/__init__.py [hunk 4]
```diff
@@ -362,6 +364,18 @@ def debug_access_token(self, token, app_id, app_secret):
...
+        """
+        url = FACEBOOK_URL + self.version + '/' + FACEBOOK_OAUTH_DIALOG_URI
```
Comment (martey) [inline facebook/__init__.py]: Can we use [`.format`](https://pyformat.info/) instead of string concatenation? See #326.

File: test/test_facebook.py [hunk 5]
```diff
@@ -151,14 +154,22 @@ def test_invalid_format(self):
...
-        actual_url = facebook.auth_url(self.app_id, redirect_url, perms=perms)
+        expected_url = ''.join([
+            facebook.FACEBOOK_URL,
+            'v' + default_version + '/',
+            facebook.FACEBOOK_OAUTH_DIALOG_URI,
+            urlencode(
+                dict(client_id=self.app_id,
+                     redirect_uri=redirect_url,
+                     scope=','.join(perms)))
+        ])
```
Comment (martey) [inline test/test_facebook.py]: I think this should use `.format` instead of `.join` and `api.version` instead of `default_version` (which doesn't need to exist for reasons I outlined earlier).

File: facebook/__init__.py [hunk 6]
```diff
@@ -44,7 +44,9 @@
 __version__ = version.__version__
 
 FACEBOOK_GRAPH_URL = "https://graph.facebook.com/"
-FACEBOOK_OAUTH_DIALOG_URL = "https://www.facebook.com/dialog/oauth?"
+FACEBOOK_URL = "https://www.facebook.com/"
+FACEBOOK_OAUTH_DIALOG_URI = "dialog/oauth?"
+FACEBOOK_OAUTH_DIALOG_URL = FACEBOOK_URL + FACEBOOK_OAUTH_DIALOG_URI
```
Comment (mikenai) [inline facebook/__init__.py]: I try to keep it for backward capability. It not used anywhere, I will remove it

File: test/test_facebook.py (hunk 3 above)
Comment (mikenai) [inline test/test_facebook.py]: I try to make the test more clear to understand why we didn`t specify API version but get a link with version 2.4.

File: test/test_facebook.py (hunk 5 above)
Comment (mikenai) [inline test/test_facebook.py]: ok, will fix that

File: facebook/__init__.py (hunk 4 above)
Comment (mikenai) [inline facebook/__init__.py]: yes we can

File: facebook/__init__.py (hunk 1 above)
Comment (mikenai) [inline facebook/__init__.py]: I think would be better to keep docstring without a link to Facebook Doc, better to keep it at project documentation because it could be changed anytime.
I will take your suggestion of that docstring - "Build a URL to create an OAuth dialog".

**Comment by mikenai** [comment]
@martey Thank you for your suggestion. Welcome to review new changes.

**Comment by HarshilPatel007** [comment]
Hey Guys, You Done a Great Job.
I'll Waiting For Graph 2.9 Support. In sort, Stable Version Of Facebook-sdk.

**Comment by benjawan13071995** [comment]
http://run.plnkr.co/preview/cj8cbz5qb00083t5yta2r9kzb/

**Comment by mikenai** [comment]
how can be failed test which is already merged to master branch?

**Comment by martey** [comment]
@mikenai I am aware of the broken test in `test_extend_access_token`; I will fix it when I have time. You can ignore it for purposes of this pull request.
